externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/KyuubiSparkUtil.scala

```diff
@@ -48,7 +48,7 @@ object KyuubiSparkUtil extends Logging {
         interruptOnCancel = true)
       debug(s"Execute initialization sql: $sql")
       try {
-        spark.sql(sql).isEmpty
+        spark.sql(sql).take(1).isEmpty
```

Comment:
`spark.sql(sql).take(1)` has two issues:
- it will do an row conversion from `InternalRow` to `Row`
- it does not do column pruning

how about using the same way of the pr https://github.com/apache/spark/pull/45373 to improve it ?

Reply:
Not sure whether it is easy to maintain this workaround in kyuubi end.